display/d.labels/do_labels.c

```diff
@@ -67,7 +71,15 @@ int initialize_options(void)
 
 int do_labels(FILE *infile, int do_rotation)
 {
-    char buff[128];
+    char buff[BUFFSIZE];
+    char buff_fmt[10];
+    char font_fmt[10];
+    char word_fmt[10];
+
+    snprintf(buff_fmt, sizeof(buff_fmt), "%%%ds", BUFFSIZE - 1);
+    snprintf(font_fmt, sizeof(font_fmt), "%%%ds", FONTSIZE - 1);
```

Comment:
`"%%%ds"` will not produce the intended format.

Reply:
will "%%ds" work

Reply:
Take a good look at the original format. (You may also test this kind of changes in a test file, which can be used as "playground").

Reply:
It was %127s.
So now it should be %%%ds right
first %% will be %
%d for BUFFSIZE - 1
s for character

Reply:
It was `"%*s %s"`

Reply:
ok so will it be this "%%*s"

i am using snprintf first time so little confused trying to understand from documentation

Reply:
It was `"%*s %s"`, you originally (correctly) wanted to change this to `"%*s %127s"`, so how will the format be to produce this format dynamically?

Formatting with [printf() and co](https://en.cppreference.com/w/c/io/fprintf) is very useful, so I recommend you study it and practice with a single file.

Reply:
snprintf(buff_fmt, sizeof(buff_fmt), "%%*s %%%ds", BUFFSIZE - 1);
sscanf(text, buff_fmt, buff);

this is working. Is it correct?

Reply:
> snprintf(buff_fmt, sizeof(buff_fmt), "%%*s %%%ds", BUFFSIZE - 1); sscanf(text, buff_fmt, buff);
> 
> this is working. Is it correct?

Yes!

Reply:
Thanks for providing good documentation. I was reading wrong examples

Reply:
done